src/ansiblelint/__main__.py

```diff
@@ -132,6 +132,11 @@ def initialize_options(arguments: list[str] | None = None) -> None | FileLock:
     for k, v in new_options.__dict__.items():
         setattr(options, k, v)
 
+    if options.nodeps is None or options.nodeps is False:
```

Comment:
This is not how env vars are supposed to work. When defined, they should always override the config options.

Preference order should always be: cli > env > config.